protocol_test.go

```diff
@@ -3,12 +3,117 @@ package tarantool_test
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
 	"github.com/tarantool/go-iproto"
-
 	. "github.com/tarantool/go-tarantool/v3"
 )
 
+// MockConnection - mock connection with Tarantool
+type MockConnection struct {
+	mock.Mock
+}
+
+func (m *MockConnection) ProtocolInfo() ProtocolInfo {
+	args := m.Called()
+	return args.Get(0).(ProtocolInfo)
+}
+
+func (m *MockConnection) Close() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+func (m *MockConnection) Addr() string {
+	args := m.Called()
+	return args.String(0)
+}
+
+func TestProtocolFeaturesSyncAndInsertArrow_WithMocks11(t *testing.T) {
+	// Creating mock connections
+	mockConn := new(MockConnection)
+
+	// Configuring so that it returns ProtocolInfo with the necessary features
+	expectedProtocolInfo := ProtocolInfo{
+		Version: ProtocolVersion(6),
+		Features: []iproto.Feature{
+			iproto.IPROTO_FEATURE_STREAMS,
+			iproto.IPROTO_FEATURE_TRANSACTIONS,
+			iproto.IPROTO_FEATURE_ERROR_EXTENSION,
+			iproto.IPROTO_FEATURE_WATCHERS,
+			iproto.IPROTO_FEATURE_PAGINATION,
+			iproto.IPROTO_FEATURE_SPACE_AND_INDEX_NAMES,
+			iproto.IPROTO_FEATURE_WATCH_ONCE,
+			iproto.IPROTO_FEATURE_IS_SYNC,
+			iproto.IPROTO_FEATURE_INSERT_ARROW,
+		},
+	}
+
+	// Setting expectations for the mock
+	mockConn.On("ProtocolInfo").Return(expectedProtocolInfo)
+	mockConn.On("Close").Return(nil)
+	mockConn.On("Addr").Return("127.0.0.1:3013")
+
+	// Performing the mock test
+	protocolInfo := mockConn.ProtocolInfo()
+
+	// Check that the features are present
+	assert.Contains(t, protocolInfo.Features, iproto.IPROTO_FEATURE_IS_SYNC,
+		"IPROTO_FEATURE_IS_SYNC should be supported")
+	assert.Contains(t, protocolInfo.Features, iproto.IPROTO_FEATURE_INSERT_ARROW,
+		"IPROTO_FEATURE_INSERT_ARROW should be supported")
+
+	mockConn.AssertCalled(t, "ProtocolInfo")
+}
+
+// Alternative version with testing via the interface
+func TestProtocolFeaturesInClientProtocolInfo(t *testing.T) {
+	/// Testing the global clientProtocolInfo from protocol.go
+	// To do this, we need to either make it exportable or test it through a public API
+
+	// Creating an IdRequest with the current clientProtocolInfo
```

Comment:
A comment should end with a dot.
```suggestion
	// Creating an IdRequest with the current clientProtocolInfo.
```